ddtrace/bootstrap/sitecustomize.py

```diff
@@ -127,6 +127,8 @@ def update_patched_modules():
     # are initialized
     if asbool(os.getenv("DD_INSTRUMENTATION_TELEMETRY_ENABLED")):
         telemetry_writer.enable()
+    else:
+        telemetry_writer.disable()
```

Comment:
My fix to collect telemetry data before enabling should only work for integrations 
https://github.com/DataDog/dd-trace-py/blob/e000eb91bf07bc82708332c478683a1fa22de056/ddtrace/internal/telemetry/writer.py#L141-L151

For other telemetry events, the guard currently is 
https://github.com/DataDog/dd-trace-py/blob/e000eb91bf07bc82708332c478683a1fa22de056/ddtrace/internal/telemetry/writer.py#L124-L135

So it shouldn't cause a leak because by default `_enabled` is None

Reply:
can we add a code comment for why it needs to be disabled here?